### ISSUE TITLE
QRouteTab breaks when using beforeRouteLeave

### ISSUE THREAD

**Issue by GrandUltimate** [quasarframework/quasar#2709]
### Software version

Quasar: 0.17.19
OS: Ubuntu 18.04.1 LTS
Node: 8.10.0
NPM: 6.4.1
Browsers: FireFox 62.0.3 (64-bit), Chrome 70.0.3538.67

### What did you get as the error?
When using `beforeRouteLeave` and calling `next(false)` to prevent route change,
the route stays the same but the tab pane switches although the `selected` tab correctly remains.

### What were you expecting?
Tab pane should not change and should stay as pane for current route and selected tab.

### What steps did you take, to get the error?
when switching between tabs use this function.
```
 beforeRouteLeave (to, from, next) {
    const answer = window.confirm('Do you really want to leave? you have unsaved changes!')
    if (answer) {
      next()
    } else {
      next(false)
    }
  },
```

**Comment by caiotarifa** [comment]
+1

**Comment by rstoenescu** [comment]
Fixed for v1.

**Comment by namespace-github** [comment]
@rstoenescu This problem I have with Quasar 2.3.4. Can it be that the problem exists again? The next tab is marked as "selected" although the route change was prevented.

**Comment by rstoenescu** [comment]
@namespace-github can you create a codesandbox or a codepen or a repo so I can play with your setup pls?

**Comment by namespace-github** [comment]
@rstoenescu Thank you for the quick response. 

Here is the behavior in codesandbox: https://codesandbox.io/s/musing-moore-dj725

If you uncheck `Allow route change`, then the route change in the pages becomes prevented in `beforeRouteLeave`  by `next(false)` (https://router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards). Then the `<q-tabs />` will show the route change, that did not take place, and mark the wrong `<q-route-tab />` as active.

Thanks a lot for your help!

**Comment by yvhr** [comment]
I'm experiencing a related issue in v2.4.2. I'm also displaying a prompt dialog when the user clicks on a route-tab item that requires authentication, prompting them to log in first, when they hit cancel, I prevent the route navigation. Router works as expected, QRouteTabs on the other hand, displays the wrong active state.

Does the `QRouteTab` track a different active state other than the current route? Or is that not a good idea?

**Comment by hendrik-schneider** [comment]
Hello to all, we have the same problem that @namespace-github wrote with Quasar 2.3.4.

**Comment by namespace-github** [comment]
@rstoenescu The Problem still exists in Quasar 2.5.2

> Here is the behavior in codesandbox: https://codesandbox.io/s/musing-moore-dj725

**Comment by rstoenescu** [comment]
@namespace-github @vycoder 
A new ticket should have been opened for this on Qv2.
Regardless, fix will be available in Quasar v2.5.4

**Comment by namespace-github** [comment]
Yes sir, thanks a lot 👍
